daemon/api.go

```diff
@@ -1673,6 +1680,41 @@ func getPaymentMethods(c *Command, r *http.Request, user *auth.UserState) Respon
 	return SyncResponse(paymentMethods, nil)
 }
 
+func readyToBuy(c *Command, r *http.Request, user *auth.UserState) Response {
+	s := getStore(c)
+
+	err := s.ReadyToBuy(user)
+
+	switch err {
+	default:
+		return InternalError("%v", err)
+	case store.ErrInvalidCredentials:
+		return Unauthorized(err.Error())
+	case store.ErrTosNotAccepted:
+		return SyncResponse(&resp{
+			Type: ResponseTypeError,
+			Result: &errorResult{
+				Message: err.Error(),
+				Kind:    errorKindTosNotAccepted,
+			},
+			Status: http.StatusBadRequest,
+		}, nil)
+	case store.ErrNoPaymentMethods:
+		return SyncResponse(&resp{
+			Type: ResponseTypeError,
+			Result: &errorResult{
+				Message: err.Error(),
+				Kind:    errorKindNoPaymentMethods,
+			},
+			Status: http.StatusBadRequest,
+		}, nil)
+	case nil:
+		// continue
+	}
+
+	return SyncResponse(true, nil)
```

Comment:
yeah, true seems fine. Alternatively `nil`, but it seems like a poor reply to a “are you ready” kind of question.